atlp-rwanda/atlp-devpulse-bn
# A user should be able to comment and react to a blog

**### PR Description**

**Feature:** A user should be able to comment and react to blog posts.

**What does this PR do?**

This PR introduces functionality allowing users to:

- Add comments to a blog post.
- Fetch all comments related to a given blog post.
- React to blog posts with reactions (e.g., Like, Love, Support, Funny).
- Like individual comments.
- Add replies to specific comments.

**Related Issues**
N/A

**Changes Made:**

- Added a model to handle reactions at the database level.
- Created a schema to define the GraphQL schema for comments.
- Created src/schema/reactionSchema.ts to define the GraphQL schema for reactions.
- Added a resolver for creating comments.
- Added a resolver for managing likes on comments.
- Added a resolver for for managing reactions on both blog posts.

**How to test this PR**

- Clone the repository
- Run npm install to install dependencies and navigate to the ft-comments-reactions-bn branch.
- Start the development server using npm run dev.
- Perform any reaction or comment query or mutation and verify the changes reflect in the database.


**Screenshoots**

<img width="949" alt="image" src="https://github.com/user-attachments/assets/cd3dd5bb-d1f2-469d-9cd6-11ef0a0fb752">
<img width="634" alt="image" src="https://github.com/user-attachments/assets/0b55fb38-13da-4432-bce9-9c6dec79ad65">
<img width="629" alt="image" src="https://github.com/user-attachments/assets/2849d4ff-d087-46c8-908a-34924b10c381">
<img width="630" alt="image" src="https://github.com/user-attachments/assets/dc5f9c1d-5a6a-4d2b-9a1d-ac9e1e429eb3">
